### ISSUE TITLE
Android Client does not support Self-Signed Certificates

### ISSUE THREAD

**Issue by MJPGPleasant** [RocketChat/Rocket.Chat.ReactNative#2682]
### Description: 

Android version of Rocket.Chat does not allaw you to accept Self signed or otherwise invalid certificates.

### Environment Information:

- Rocket.Chat Server Version: 3.8.0
- Rocket.Chat App Version: 4.12.1
- Device Name: Samsung J3 (2016)
- OS Version: 7.1.1

### Steps to reproduce:

1. Configure a Rocket.Chat server using a self-signed certificate as per published Rocket.Chat instructions (Docker Install / Nginx)  
2. Install Rocket.Chat Android App from Google Store
3. Access the server in step one with the client in step two

### Expected behavior:

A prompt similar to the Desktop Client stating that the certificate is invalid and prompting if you want to accept the certificate

### Actual behavior:

Oops!

java.security.cert.CertPathValidatorExeception: Trust anchor for certification path not found.

**Comment by MJPGPleasant** [comment]
I have successfully installed the certificate, but am having difficulties completing the trust.  RocketChat now shows:

Oops!

Hostname <fqdn> not verified: 
     certificate: sha256/<certhash>
     DN: <Dn text>
    subjectAltNames: []

Let me know if you can supply any more hints as to how to go about installing the certificate.

**Comment by diegolmello** [comment]
@MJPGPleasant On my Android, I have to search for `security certificates` on Settings and check for `User` certificates.
It was disabled.
There're a couple of tutorials on Google telling how to do it.
I couldn't find a good one.
Let me know if it worked.

**Comment by MJPGPleasant** [comment]
@diegolmello I have the certificate listed under `View security certificates` > `user` But see no way to enable or disable. Not sure if this is a limitation of the phone or the version of Android I am using. 

I would like to see if I could get this working, but at this point the requirements are quickly surpassing what we can expect our end clients to do to set up their own devices.

Thanks for the help, and let me know if you have any further suggestions as to what we might need to get this functional.

**Comment by diegolmello** [comment]
To check if the certificate is working, try accessing your server on your device's browser.

**Comment by MJPGPleasant** [comment]
Chrome browser has an alert (!) in the address bar, but otherwise allows me through to the site with no additional prompts or other pages before the main site. Not sure if this is because I have previously accepted the certificate in this browser. 

The Samsung Browser brings up a standard `Your connection is not private` page, with you having to go to `Advanced` > 'Proceed to <server>` to advance to the logon page.

**Comment by diegolmello** [comment]
@MJPGPleasant yeah, that's a sign showing the certificate isn't installed (working).
Browsers let you access insecure websites normally, but tell you if it's encrypted or not.
Mobile apps following security best practices like ours, only accept secure connections.
That's why your server isn't working.

**Comment by MJPGPleasant** [comment]
Agreed - The certificate is installed (changing the RC error from `Trust anchor for certification path not found` to `Hostname not verified` However the certificate is still not being trusted by Android, either because of configuration of the cert or the target phone. 

I will do some more digging through the Android forums to see if I can find anything. In the meantime if you have any guesses let me know. If I find anything I will likewise post as it might be handy for anyone else who comes across this.

**Comment by MJPGPleasant** [comment]
Upon further investigation there was other un-set parameters in the self-signed certificate that Android requires. We have decided to get a properly secured Cert from a CA to eliminate this and other workarounds required for Self Signed Certs vs investigating this further.

**Comment by adrastest** [comment]
@MJPGPleasant  could you find any solution for hostname not verified?
if yes plz share to solution i have the same problem.
